src/HDF5.jl

```diff
@@ -971,19 +983,62 @@ end
 t_commit(parent::Union{HDF5File, HDF5Group}, path::String, dtype::HDF5Datatype) = t_commit(parent, path, dtype, p_create(H5P_LINK_CREATE))
 
 a_create(parent::Union{HDF5File, HDF5Object}, name::String, dtype::HDF5Datatype, dspace::HDF5Dataspace) = HDF5Attribute(h5a_create(checkvalid(parent).id, name, dtype.id, dspace.id), file(parent))
+
+function _prop_set!(p::HDF5Properties, name::Symbol, val, check::Bool = true)
+    class = p.class
+
+    if class == H5P_FILE_CREATE
+        return name === :userblock   ? h5p_set_userblock(p, val...) :
+               name === :track_times ? h5p_set_obj_track_times(p, val...) : # H5P_OBJECT_CREATE
+               check ? error("unknown file create property ", name) : nothing
+    end
+
+    if class == H5P_FILE_ACCESS
+        return name === :alignment     ? h5p_set_alignment(p, val...) :
+               name === :fclose_degree ? h5p_set_fclose_degree(p, val...) :
+               name === :libver_bounds ? h5p_set_libver_bounds(p, val...) :
+               check ? error("unknown file access property ", name) : nothing
+    end
+
+    if class == H5P_GROUP_CREATE
+        return name === :local_heap_size_hint ? h5p_set_local_heap_size_hint(p, val...) :
+               name === :track_times          ? h5p_set_obj_track_times(p, val...) : # H5P_OBJECT_CREATE
+               check ? error("unknown group create property ", name) : nothing
+    end
+
+    if class == H5P_LINK_CREATE
+        return name === :char_encoding ? h5p_set_char_encoding(p, val...) :
+               name === :create_intermediate_group ? h5p_set_create_intermediate_group(p, val...) :
+               check ? error("unknown link create property ", name) : nothing
+    end
+
+    if class == H5P_DATASET_CREATE
+        return name === :alloc_time  ? h5p_set_alloc_time(p, val...) :
+               name === :blosc       ? h5p_set_blosc(p, val...) :
+               name === :chunk       ? set_chunk(p, val...) :
+               name === :compress    ? h5p_set_deflate(p, val...) :
+               name === :deflate     ? h5p_set_deflate(p, val...) :
+               name === :external    ? h5p_set_external(p, val...) :
+               name === :layout      ? h5p_set_layout(p, val...) :
+               name === :shuffle     ? h5p_set_shuffle(p, val...) :
+               name === :track_times ? h5p_set_obj_track_times(p, val...) : # H5P_OBJECT_CREATE
+               check ? error("unknown dataset create property ", name) : nothing
+    end
+
+    if class == H5P_ATTRIBUTE_CREATE
+        return name === :char_encoding ? h5p_set_char_encoding(p, val...) :
+               check ? error("unknown attribute create property ", name) : nothing
+    end
+
+    check ? error("unknown property class ", class) : return nothing
```

Comment:
```suggestion
    return check ? error("unknown property class ", class) : nothing
```

I guess these could also just be a series of if .. ifelse.. although that doesn't matter